convert2rhel/unit_tests/subscription_test.py

```diff
@@ -1266,3 +1267,80 @@ def test_update_rhsm_custom_facts_disable_telemetry(monkeypatch, caplog):
     subscription.update_rhsm_custom_facts()
 
     assert message in caplog.records[-1].message
+
+
+def test_get_rhsm_facts(monkeypatch, global_tool_opts, tmpdir):
+    facts_string = {
+        "cpu.cpu(s)": "8",
+        "cpu.cpu_socket(s)": "3",
+    }
+    facts_json = json.dumps(facts_string)
+    facts_file = tmpdir.join("facts.json")
+    facts_file.write(facts_json)
+    monkeypatch.setattr(subscription, "tool_opts", global_tool_opts)
+    monkeypatch.setattr(subscription, "RHSM_FACTS_FILE", str(facts_file))
+    global_tool_opts.no_rhsm = False
+    facts = subscription.get_rhsm_facts()
+    assert facts == {"cpu.cpu(s)": "8", "cpu.cpu_socket(s)": "3"}
+
+
+def test_get_rhsm_facts_no_rhsm(monkeypatch, global_tool_opts):
+    run_mock = RunSubprocessMocked(return_string="")
+    monkeypatch.setattr(subscription, "tool_opts", global_tool_opts)
+    global_tool_opts.no_rhsm = True
+
+    facts = subscription.get_rhsm_facts()
+
+    assert isinstance(facts, dict)
+    assert len(facts) == 0
+    assert run_mock.call_count == 0
+
+
+def test_get_rhsm_facts_file_not_found(monkeypatch, global_tool_opts, caplog):
+    run_mock = RunSubprocessMocked(return_string="")
+    monkeypatch.setattr(subscription, "tool_opts", global_tool_opts)
+    monkeypatch.setattr(subscription, "RHSM_FACTS_FILE", "fake/location/file.json")
+    global_tool_opts.no_rhsm = False
+    facts = subscription.get_rhsm_facts()
+    assert caplog.records[-1].levelname == "CRITICAL"
+    assert (
+        caplog.records[-1].message
+        == "Failed to get the RHSM facts : [Errno 2] No such file or directory: 'fake/location/file.json'."
+    )
+    assert isinstance(facts, dict)
+    assert len(facts) == 0
+    assert run_mock.call_count == 0
+
+
+@centos7
+@pytest.mark.skipif(pkgmanager.TYPE == "dnf", reason="Test is only relevant for RHEL 7")
+def test_get_rhsm_facts_json_decode_error_el7(monkeypatch, global_tool_opts, caplog, tmpdir, pretend_os):
+    run_mock = RunSubprocessMocked(return_string="")
+    facts_string = "cpu.cpu(s) :8 "
+    facts_file = tmpdir.join("facts.json")
+    facts_file.write(facts_string)
+    monkeypatch.setattr(subscription, "tool_opts", global_tool_opts)
+    monkeypatch.setattr(subscription, "RHSM_FACTS_FILE", str(facts_file))
+    facts = subscription.get_rhsm_facts()
+    assert caplog.records[-1].levelname == "CRITICAL"
+    assert caplog.records[-1].message == "Failed to get the RHSM facts : No JSON object could be decoded."
+    assert isinstance(facts, dict)
+    assert len(facts) == 0
+    assert run_mock.call_count == 0
+
+
+@centos8
+@pytest.mark.skipif(pkgmanager.TYPE == "yum", reason="Test is only relevant for RHEL 8+")
+def test_get_rhsm_facts_json_decode_error_el7(monkeypatch, global_tool_opts, caplog, tmpdir, pretend_os):
```

Comment:
Could we update the name in here to el8?

```suggestion
def test_get_rhsm_facts_json_decode_error_el8(monkeypatch, global_tool_opts, caplog, tmpdir, pretend_os):
```